contrib/internal/httptrace/httptrace.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package httptrace provides functionalities to trace HTTP requests that are commonly required and used across
+// contrib/** integrations.
+package httptrace
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// List of standard HTTP request span tags.
+const (
+	// HTTPMethod is the HTTP request method.
+	HTTPMethod = ext.HTTPMethod
+	// HTTPURL is the full HTTP request URL in the form `scheme://host[:port]/path[?query][#fragment]`.
+	HTTPURL = ext.HTTPURL
+	// HTTPUserAgent is the user agent header value of the HTTP request.
+	HTTPUserAgent = "http.useragent"
+	// HTTPCode is the HTTP response status code sent by the HTTP request handler.
+	HTTPCode = ext.HTTPCode
+)
+
+// StartRequestSpan starts an HTTP request span with the standard list of HTTP request span tags. URL query parameters
+// are added to the URL tag when queryParams is true. Any further span start option can be added with opts.
+func StartRequestSpan(r *http.Request, queryParams bool, opts ...ddtrace.StartSpanOption) (tracer.Span, context.Context) {
```

Comment:
Internal or not, I think `bool` args are not a good programming practice. There is only one single place in all of the usage in this repository where the `queryParams` param has any other value than `false`. I don't think that this one place warrants introducing a new parameter and an additional helper function. It isn't really de-duplicating anything.

Can we please get rid of the `queryParams` parameter and the `makeURLTag` helper? Instead, simply add the `tracer.Tag(HTTPURL, ...)` to `opts` in `net/http` when needed.

Reply:
Actually it seems like we've already been through this https://github.com/DataDog/dd-trace-go/pull/1286#discussion_r873836690 🙂

Reply:
> There is only one single place in all of the usage in this repository where the queryParams param has any other value than false.

If we follow the specification as specified today, we could remove this argument because the query params are supposed to be traced by default now, with an opt-out env var that we could check inside `httptrace`.

I wouldn't move the `http.url` tag creation out of `httptrace` to avoid the former copy/paste issue, which is the only purpose of this PR.